Sources/Jupiter/JupiterRestClientAPI.swift

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 P2P Validator Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import SolanaSwift
+
+public class JupiterRestClientAPI: JupiterAPI {
+    private let host: String
+
+    public init(version: Version) {
+        host = "https://quote-api.jup.ag/" + version.rawValue
+    }
+    
+    public func getTokens() async throws -> [Token] {
+        let (data, _) = try await URLSession.shared.data(from: URL(string: "https://cache.jup.ag/tokens")!)
+        return try JSONDecoder().decode([Token].self, from: data)
+    }
+
+    public func quote(
+        inputMint: String,
+        outputMint: String,
+        amount: String,
+        swapMode: SwapMode?,
+        slippageBps: Int?,
+        feeBps: Int?,
+        onlyDirectRoutes: Bool?,
+        userPublicKey: String?,
+        enforceSingleTx: Bool?
+    ) async throws -> Response<[Route]> {
+        guard var urlComponent = URLComponents(string: host + "/quote") else { throw JupiterError.invalidURL }
+
+        // Queries
+        var queries: [URLQueryItem] = [
+            .init(name: "inputMint", value: inputMint),
+            .init(name: "outputMint", value: outputMint),
+            .init(name: "amount", value: amount),
+            .init(name: "userPublicKey", value: userPublicKey),
+        ]
+
+        if let swapMode = swapMode {
+            queries.append(.init(name: "swapMode", value: swapMode.rawValue))
+        }
+        if let slippageBps = slippageBps {
+            queries.append(.init(name: "slippageBps", value: String(slippageBps)))
+        }
+        if let onlyDirectRoutes = onlyDirectRoutes {
+            queries.append(.init(name: "onlyDirectRoutes", value: String(onlyDirectRoutes)))
+        }
+        if let enforceSingleTx = enforceSingleTx {
+            queries.append(.init(name: "enforceSingleTx", value: String(enforceSingleTx)))
+        }
+        if let feeBps = feeBps {
+            queries.append(.init(name: "feeBps", value: String(feeBps)))
+        }
+        urlComponent.queryItems = queries
+
+        guard let url = urlComponent.url else { throw JupiterError.invalidURL }
+        let request = URLRequest(url: url)
+
+        let (data, _) = try await URLSession.shared.data(for: request)
+        print(request.cURL())
+        print(String(data: data, encoding: .utf8))
+        return try JSONDecoder().decode(Response<[Route]>.self, from: data)
+    }
+
+    public func swap(
+        route: Route,
+        userPublicKey: String,
+        wrapUnwrapSol: Bool,
+        feeAccount: String?,
+        asLegacyTransaction: Bool?,
+        computeUnitPriceMicroLamports: Int?,
+        destinationWallet: String?
+    ) async throws -> Transaction? {
```

Comment:
Is it possible to make Transaction, not optional?

guard let transaction else { throw SomeDecodingError }